cmd/kubeops/main_test.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kubeapps/kubeapps/cmd/kubeops/internal/handler"
+)
+
+func TestParseAdditionalClusterConfig(t *testing.T) {
+	testCases := []struct {
+		name           string
+		configJSON     string
+		expectedErr    bool
+		expectedConfig map[string]handler.AdditionalClusterConfig
+	}{
+		{
+			name:       "parses a single additional cluster",
+			configJSON: `[{"name": "cluster-2", "apiServiceURL": "https://example.com", "certificateAuthorityData": "abcd"}]`,
+			expectedConfig: map[string]handler.AdditionalClusterConfig{
+				"cluster-2": handler.AdditionalClusterConfig{
+					Name:                     "cluster-2",
+					ApiServiceURL:            "https://example.com",
+					CertificateAuthorityData: "abcd",
+				},
+			},
+		},
+		{
+			name: "parses multiple additional clusters",
+			configJSON: `[
+	{"name": "cluster-2", "apiServiceURL": "https://example.com/cluster-2", "certificateAuthorityData": "abcd"},
+	{"name": "cluster-3", "apiServiceURL": "https://example.com/cluster-3", "certificateAuthorityData": "efgh"}
+]`,
+			expectedConfig: map[string]handler.AdditionalClusterConfig{
+				"cluster-2": handler.AdditionalClusterConfig{
+					Name:                     "cluster-2",
+					ApiServiceURL:            "https://example.com/cluster-2",
+					CertificateAuthorityData: "abcd",
+				},
+				"cluster-3": handler.AdditionalClusterConfig{
+					Name:                     "cluster-3",
+					ApiServiceURL:            "https://example.com/cluster-3",
+					CertificateAuthorityData: "efgh",
+				},
+			},
+		},
+		{
+			name:        "errors if the cluster configs cannot be parsed",
+			configJSON:  `[{"name": "cluster-2", "apiServiceURL": "https://example.com", "certificateAuthorityData": "extracomma",}]`,
+			expectedErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			path := createConfigFile(t, tc.configJSON)
+			defer os.Remove(path)
+
+			config, err := parseAdditionalClusterConfig(path)
+			if got, want := err != nil, tc.expectedErr; got != want {
+				t.Errorf("got: %t, want: %t", got, want)
+			}
+
+			if got, want := config, tc.expectedConfig; !cmp.Equal(want, got) {
+				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got))
+			}
+		})
+	}
+}
+
+func createConfigFile(t *testing.T, content string) string {
```

Comment:
To avoid having to create temporary files, I would move the `ioutil.ReadFile` action outside the `parse` function. It's not a big deal but this would prevent the tests from running in a read-only environment like a container.

Reply:
Yeah - even when running on a read-only container we can mount a `tmpfs` aiui. The reason I include as much as possible in the function rather than in the main is exactly so we're testing as much as we can. CLI passes a path, this function verifies that it can return the config when given that path.